tools/src/main/scala/scala/scalanative/interflow/Combine.scala

```diff
@@ -0,0 +1,615 @@
+package scala.scalanative
+package interflow
+
+import scalanative.nir._
+import scalanative.linker._
+import scalanative.util.{unreachable, And}
+import nir.Bin.{And => Iand, _}
+import nir.Comp._
+import nir.Conv._
+
+trait Combine { self: Interflow =>
```

Comment:
@sjrd It would be great if you could give it a brief look in case anything looks obviously wrong.